ENH: Adds node labels to targets

thomasjpfan opened the pull request:
Closes #42.

1. Adds node labels to targets.
2. Updates flexible label documentation to take advantage of this feature.
3. Adds endpoints `/v1/docker-flow-monitor/node/reconfigure` and `/v1/docker-flow-monitor/node/remove` to listen for node events.

@vfarcic I added the env variable `DF_GET_NODES_URL` to allow the user to define the url to get nodes information from during initial configuration. If you disagree with this additional variable, I could use the `LISTENER_ADDRESS` env variable and hard code the path in.

thomasjpfan commented:
@vfarcic Are you okay with this PR?

vfarcic commented on docs/config.md:
```diff
@@ -110,6 +110,11 @@ Additional scrapes can be added through files prefixed with `scrape_`. By defaul
-When using a version of [Docker Flow Swarm Listener](https://github.com/vfarcic/docker-flow-swarm-listener), DFSL, newer than `18.02.06-31`, you can configure DFSL to send node node hostnames to `Docker Flow Monitor`, DFM. This can be done by setting `DF_INCLUDE_NODE_IP_INFO` to `true` in the DFSL environment. DFM will automatically display the node hostnames as a label for each prometheus target ...
+When using a version of [Docker Flow Swarm Listener](https://github.com/vfarcic/docker-flow-swarm-listener), DFSL, newer than `18.03.20-39`, you can configure DFSL to send node information to `Docker Flow Monitor`, DFM. This can be done by setting `DF_INCLUDE_NODE_IP_INFO` to `true` in the DFSL environment. DFM will automatically display the node hostnames as a label for each prometheus target. T ...
+
+In addition to service labels, DFM can be configured to import node and engine labels prefixed with `com.df.` as prometheus labels for our targets. First, configure DFSL to push node events to DFM by setting `DF_NOTIFY_CREATE_NODE_URL=[MONITOR_IP]:[MONITOR_PORT]/v1/docker-flow-monitor/node/reconfigure` and `DF_NOTIFY_REMOVE_NODE_URL=[MONITOR_IP]:[MONITOR_PORT]/v1/docker-flow-monitor/node/remove`  ...
```
I would change `[MONITOR_IP]` to `[MONITOR_DNS]`. It might be clearer that people should not look for the actual IP of the monitor service.

vfarcic replied:
There's a typo: `Flexiable` > `Flexible`.

vfarcic replied:
Maybe add a reference to a YAML or some other path of the tutorial that has an example with `DF_NOTIFY_CREATE_NODE_URL`?

thomasjpfan commented:
1. I added more details in the configuration section by going through an example of a yaml file.
2. I changed the formatting on environmental variables to make it a little more readable.

vfarcic commented:
I just realized that I forgot to add you to collaborators. Now you can merge PRs (yours or from others).
